backend/src/test/java/org/uresti/pozarreal/service/impl/CourseAssistantPaymentServiceImplTests.java

```diff
@@ -0,0 +1,219 @@
+                .id("id1")
+                .paymentDate(now)
+                .amount(new BigDecimal(50))
+                .courseAssistantId("abc")
+                .receiptNumber("12345")
+                .notes("hello")
+                .userId("User 1")
+                .concept("Concept 1")
+                .build();
+
+        CoursePayment coursePaymentDto = CoursePayment.builder()
+                .paymentDate(now)
+                .amount(new BigDecimal(60))
+                .courseAssistantId("abcd")
+                .receiptNumber("123456")
+                .notes("hello2")
+                .userId("User 2")
+                .concept("Concept 2")
+                .build();
+
+        Mockito.when(courseAssistantPaymentRepository.save(argumentCaptor.capture())).thenReturn(coursePayment);
+
+        // When:
+        CoursePayment courseAssistant = courseAssistantPaymentService.save(coursePaymentDto);
+
+        // Then:
+        org.uresti.pozarreal.model.CoursePayment parameter = argumentCaptor.getValue();
+
+        Assertions.assertThat(parameter.getId()).isNotNull();
+        Assertions.assertThat(parameter.getPaymentDate()).isEqualTo(now);
+        Assertions.assertThat(parameter.getUserId()).isEqualTo("User 2");
+        Assertions.assertThat(parameter.getCourseAssistantId()).isEqualTo("abcd");
+        Assertions.assertThat(parameter.getReceiptNumber()).isEqualTo("123456");
+        Assertions.assertThat(parameter.getAmount()).isEqualTo(new BigDecimal(60));
+        Assertions.assertThat(parameter.getNotes()).isEqualTo("hello2");
+        Assertions.assertThat(parameter.getConcept()).isEqualTo("Concept 2");
+
+        Assertions.assertThat(courseAssistant.getPaymentDate()).isEqualTo(now);
+        Assertions.assertThat(courseAssistant.getId()).isEqualTo("id1");
+        Assertions.assertThat(courseAssistant.getUserId()).isEqualTo("User 1");
+        Assertions.assertThat(courseAssistant.getCourseAssistantId()).isEqualTo("abc");
+        Assertions.assertThat(courseAssistant.getReceiptNumber()).isEqualTo("12345");
+        Assertions.assertThat(courseAssistant.getAmount()).isEqualTo(new BigDecimal(50));
+        Assertions.assertThat(courseAssistant.getNotes()).isEqualTo("hello");
+        Assertions.assertThat(courseAssistant.getConcept()).isEqualTo("Concept 1");
+
+        Assertions.assertThat(courseAssistant.getId()).isNotEqualTo(parameter.getId());
+    }
+
+    @Test
+    public void givenAnCourseAssistantPaymentId_whenDelete_thenIsDeleted() {
+        // Given:
+        CourseAssistantPaymentRepository courseAssistantPaymentRepository = Mockito
+                .mock(CourseAssistantPaymentRepository.class);
+
+        CourseAssistantPaymentServiceImpl courseAssistantPaymentService = new
+                CourseAssistantPaymentServiceImpl(courseAssistantPaymentRepository);
+
+        // When:
+        // Then:
+        courseAssistantPaymentService.delete("abc");
```

Comment:
Move this line to the section "When:" .. there's no reason to put together "When" and "Then" in this test

Reply:
done